mysqld_exporter.go

```diff
@@ -937,35 +937,38 @@ func scrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 	}
 	defer slaveStatusRows.Close()
 
-	if slaveStatusRows.Next() {
-		// There is either no row in SHOW SLAVE STATUS (if this is not a
-		// slave server), or exactly one. In case of multi-source
-		// replication, things work very much differently. This code
-		// cannot deal with that case.
-		slaveCols, err := slaveStatusRows.Columns()
-		if err != nil {
-			return err
-		}
+	// Don't bother if there is no slave status row
+	if !slaveStatusRows.Next() {
+		return nil
```

Comment:
Sorry, that's not really possible.  This scrape function works off of `SHOW SLAVE STATUS` which returns an empty set result with no column headers so we have no idea what metrics are there or not.

If you look at the change more closely, there is no functional difference, this just allows us to remove one level of indent, and abort early.